src/main/java/com/google/sps/database/RentPostDatabase.java

```diff
@@ -0,0 +1,28 @@
+package com.google.sps;
+
+import static com.google.sps.Constants.RENT_COLLECTION_NAME;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.FieldValue;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.cloud.FirestoreClient;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.ServletException;
+
+/** Handles database interaction (add, fetch) for rent posts. */
+public class RentPostDatabase {
+  /**
+  * Add a rent post document to a collection using a map.
+  * @param post New rent post. 
+  */
+  public void addPost(RentPost post) {
```

Comment:
Updating and deleting is a planned feature. Firestore automatically generates a unique id for each document if it is not specified (what we're using right now). Added a documentID optional as a variable to the RentPost (now called listing) since when pushing a document we don't have the ID, but when fetching it we do.